src/backend/joanie/tests/payment/test_backend_payplug.py

```diff
@@ -889,3 +890,222 @@ def test_payment_backend_payplug_abort_payment_request_failed(
+                    "due_date": "2024-04-17",
+                    "state": enums.PAYMENT_STATE_PENDING,
+                },
+            ],
+        )
+        backend = PayplugBackend(self.configuration)
+        mock_treat.return_value = PayplugFactories.PayplugPaymentFactory(
+            id=payment_id,
+            failure=True,
+            metadata={
+                "order_id": str(order.id),
+                "installment_id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
+            },
+        )
+
+        request = APIRequestFactory().post(
+            reverse("payment_webhook"), data={"id": payment_id}, format="json"
+        )
+
+        backend.handle_notification(request)
+
+        mock_send_mail_refused_debit.assert_called_once_with(
+            order, "d9356dd7-19a6-4695-b18e-ad93af41424a"
+        )
+
+    @mock.patch.object(payplug.notifications, "treat")
+    def test_payment_backend_payplug_send_email_about_refused_installment_payment_in_english(
+        self, mock_treat
+    ):
+        """
+        When backend receives a payment notification which failed, the generic method
+        `_do_on_payment_failure` should be called and should send an email mentioning about
+        the refused debit on the installment in the user's preferred language.
+        """
+        payment_id = "pay_failure"
+        user = UserFactory(
+            first_name="John",
+            last_name="Doe",
+            language="en-us",
+            email="john.doe@acme.org",
+        )
+        product = ProductFactory(price=D("999.99"), title="Product 1")
+        order = OrderGeneratorFactory(
+            state=ORDER_STATE_PENDING, owner=user, product=product
+        )
+        # Force the first installment id to match the stored request
+        first_installment = order.payment_schedule[0]
+        first_installment["id"] = "d9356dd7-19a6-4695-b18e-ad93af41424a"
+        order.save()
+        backend = PayplugBackend(self.configuration)
+
+        mock_treat.return_value = PayplugFactories.PayplugPaymentFactory(
+            id=payment_id,
+            failure=True,
+            metadata={
+                "order_id": str(order.id),
+                "installment_id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
+            },
+        )
+
+        request = APIRequestFactory().post(
+            reverse("payment_webhook"), data={"id": payment_id}, format="json"
+        )
+
+        backend.handle_notification(request)
+
+        email_content = " ".join(mail.outbox[0].body.split())
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertEqual(mail.outbox[0].to[0], "john.doe@acme.org")
+        self.assertIn(
+            "An installment debit has failed",
+            mail.outbox[0].subject,
+        )
+        self.assertIn("Product 1", email_content)
+
+    @mock.patch.object(payplug.notifications, "treat")
+    def test_payment_backend_payplug_send_email_about_refused_installment_in_user_language(
```

Comment:
This method name and the previous one is confusing. In both cases you test the language defined by the user. IMO for this one you should mentin the french language instead of `in_user_lanaguage` because it is already what you are testing in the previous method.